src/Models/CMAC.cpp

```diff
@@ -66,22 +66,23 @@ std::vector<std::size_t> CMACMap::getIndizes(blas::matrix_row< const RealMatrix>
 CMACMap::CMACMap():m_tilings(0) {
 	m_features|=HAS_FIRST_PARAMETER_DERIVATIVE;
 }
-void CMACMap::setStructure(std::size_t inputs, std::size_t outputs, std::size_t numberOfTilings, std::size_t numberOfTiles, double lower, double upper,bool randomTiles){
-	RealMatrix bounds(inputs,2);
-	for (std::size_t dim=0; dim != inputs; ++dim) {
+void CMACMap::setStructure(Shape const& inputs, Shape const& outputs, std::size_t numberOfTilings, std::size_t numberOfTiles, double lower, double upper,bool randomTiles){
+	RealMatrix bounds(inputs.numElements(),2);
+	for (std::size_t dim=0; dim != bounds.size1(); ++dim) {
 		bounds(dim, 0) = lower;
 		bounds(dim, 1) = upper;
 	}
 	setStructure(inputs, outputs, numberOfTilings, numberOfTiles,bounds,randomTiles);
 }
-void CMACMap::setStructure(std::size_t inputs, std::size_t outputs, std::size_t numberOfTilings, std::size_t numberOfTiles, RealMatrix const& bounds, bool randomTiles){
-	m_inputSize  = inputs;
-	m_outputSize = outputs;
+void CMACMap::setStructure(Shape const& inputs, Shape const& outputs, std::size_t numberOfTilings, std::size_t numberOfTiles, RealMatrix const& bounds, bool randomTiles){
+	m_inputSize  = inputs.numElements();
```

Comment:
m_inputSize and m_inputShape are highly redundant, in particular since the shape already keeps the size as a redundant data member. Can we remove m_inputSize?